src/ip/mod.rs

```diff
@@ -8,41 +12,26 @@ type NetworkValues = (NetworkAddr<u32>, BroadcastAddr<u32>);
 
 pub fn is_valid(address: &str) -> bool {
     let octets = address.split(".");
-    let mut octet_count: u8 = 0;
-    for byte in octets {
-        match byte.parse::<u8>() {
-            Err(_) => return false,
-            Ok(_)  => octet_count += 1,
-        }
-        if octet_count > 4 {
-            return false;
-        }
-    }
-    return octet_count == 4
+    octets.count() == 4 && make_mask_from_string(address).is_ok()
 }
 
-
 pub fn make_mask_from_cidr(cidr: u8) -> Result<u32, ()> {
     if cidr > 32 {
         return Err(());
     }
     let mask = 0xffffffff_u32 & !((0xffffffff_u64 >> cidr) as u32);   
-    return Ok(mask)
-
+    Ok(mask)
 }
 
-pub fn make_mask_from_string(address: &str) -> u32 {
-    let octets = address.split(".");
-    let mut octet_count: u8 = 3;
-    let mut mask: u32 = 0;
-    for byte in octets {
-        let ibyte = byte.parse::<u8>().unwrap();
-        mask += 256u32.pow(octet_count as u32) * ibyte as u32;
-        if octet_count > 0 {
-            octet_count -= 1;
-        }
-    }
-    return mask
+pub fn make_mask_from_string(address: &str) -> Result<u32, ParseIntError> {
+    address.split(".")
+        .enumerate()
+        .fold(Ok(0_u32), |acc, (i, octet)| {

```

Comment:
I can't find good documentation.  Is this a lambda?  Is |acc, (i, octet)| declaring the variables used within the codeblock?